Doc/library/sched.rst

```diff
@@ -19,19 +19,18 @@ scheduler:
 
    The :class:`scheduler` class defines a generic interface to scheduling events.
    It needs two functions to actually deal with the "outside world" --- *timefunc*
```

Comment:
So, this sentence isn't exactly proper English.  (I'm using the original version)

``It needs two functions to actually deal with the “outside world” — timefunc should be callable without arguments, and return a number (the “time”, in any units whatsoever). ``

Starting a sentence with `It needs two functions - ` would imply to me that I would then see those two functions after the dash and within the same sentence, which in this case would require more of a rewrite.  However, I believe this is why Victor asked for additional review because it does take a few readings of the paragraph to grasp the meaning.

Based on that, my suggestion is:

> The :class:`scheduler` class defines a generic interface to scheduling events.
  It needs two functions to actually deal with the "outside world".  *timefunc* is a 
 callable without arguments that returns a number representing the "time" (in any
  units whatsoever).  *delayfunc* is a callable with one argument, which is a 
  number compatible with the output of *timefunc*, and that delays 
  for that many  time units. *delayfunc* will also be called with the argument ``0`` after each
   event is run to allow other threads an opportunity to run in multi-threaded
   applications.

Reply:
Again, I didn't write any of the `sched` docs in the first place but am glad to help touch it up now 😆

Reply:
In 99da934, I made them more compartmentalized and also clarified the wording based off your suggestion.  I think we are all on the same page here - trying to emphasize that `delayfunc()` effectively takes the numeric output produced by `timefunc()` as its argument in doing the actual scheduling.